### DISCUSSION TITLE
Share your keybindings, shortcuts, and workflows here!

### DISCUSSION THREAD

**Discussion by AllanChain** [AllanChain/logseq-live-math#14] [Show and tell]
Logseq Live Math enables a new way of typing math in Logseq. To fully leverage this plugin, configuring your own keybindings and shortcuts are strongly encouraged. Here, we use GitHub Discussions as a platform for the users sharing the settings and learning from others. You are encouraged to share your keybindings, shortcuts, workflows, and more in this thread or in a new one.

If you need a quick settings example, [checkout this gist](https://gist.github.com/AllanChain/fa7557f7c36d04033631528f84a3b929).

**Comment by AllanChain** [comment]
## Prefer no smart fence

I don't like the `smartFence` option which automatically convert parentheses to `\left...\right` markup, because it can be too verbose. So I disabled it and added keyboard shortcuts to insert `\left...\right` pair when needed:


In `keybings` section:
```json
    {
      "key": "ctrl+shift+9",
      "command": [
        "insert",
        "\\left(#?\\right)"
      ]
    },
    {
      "key": "ctrl+[",
      "command": [
        "insert",
        "\\left[#?\\right]"
      ]
    },
    {
      "key": "ctrl+{",
      "command": [
        "insert",
        "\\left\\{#?\\right\\}"
      ]
    }
```

**Comment by AllanChain** [comment]
## Use <kbd>Enter</kbd> to accept suggestions

By default, you need `Tab` and `Enter` to confirm latex suggestion. I changed it to `Tab` to cycle suggestion and `Enter` to confirm.

In `keybindings` section:
```json
    {
      "key": "[Tab]",
      "ifMode": "latex",
      "command": "nextSuggestion"
    },
    {
      "key": "shift+[Tab]",
      "ifMode": "latex",
      "command": "previousSuggestion"
    },
    {
      "key": "[Return]",
      "ifMode": "latex",
      "command": [
        "complete",
        "accept-all"
      ]
    },
    {
      "key": "[Enter]",
      "ifMode": "latex",
      "command": [
        "complete",
        "accept-all"
      ]
    }
```
![output](https://github.com/AllanChain/logseq-live-math/assets/36528777/5740c817-37cb-4a39-8ccd-812d1fd2cf13)

**Comment by AllanChain** [comment]
## Obsidian-style Greek letters

[Obsidian Latex Suite](https://github.com/artisticat1/obsidian-latex-suite) is a great LaTeX snip plugin for obsidian. It comes with some nice defaults, including the easy input of Greek letters. I took some of them and included some others from LyX.

In `inlineShortcuts` section:
```json
    "@a": "\\alpha",
    "@A": "\\alpha",
    "@b": "\\beta",
    "@B": "\\beta",
    "@c": "\\chi",
    "@C": "\\chi",
    "@d": "\\delta",
    "@D": "\\Delta",
    "@e": "\\epsilon",
    "@E": "\\epsilon",
    ":e": "\\varepsilon",
    ":E": "\\mathcal{E}",
    "@f": "\\phi",
    "@F": "\\Phi",
    ":f": "\\varphi",
    "@g": "\\gamma",
    "@G": "\\Gamma",
    "@h": "\\eta",
    "@H": "\\eta",
    "@k": "\\kappa",
    "@K": "\\kappa",
    "@l": "\\lambda",
    "@L": "\\Lambda",
    "@m": "\\mu",
    "@M": "\\mu",
    "@n": "\\nu",
    "@N": "\\nu",
    "@o": "\\omega",
    "@O": "\\Omega",
    "@r": "\\rho",
    "@R": "\\rho",
    "@s": "\\sigma",
    "@S": "\\Sigma",
    "@t": "\\tau",
    "@T": "\\tau",
    "@u": "\\upsilon",
    "@U": "\\Upsilon",
    "ome": "\\omega",
    "@v": "\\theta",
    "@V": "\\Theta",
    "@y": "\\psi",
    "@Y": "\\Psi",
    "@x": "\\xi",
    "@X": "\\Xi",
    "@z": "\\zeta",
    "@Z": "\\zeta",
```

**Comment by AllanChain** [comment]
## Fractions without implicit arguments

By default, for example, when hitting `/` after `abc`, MathLive will regard `abc` as implicit argument and insert `\frac{abc}{}`. However, in many cases, this is not intended, for example I just need `ab\frac{c}{}` or `abc\frac{}{}`. It's possible to disable this behavior and have more control over fractions.

In `keybindings` section:
```json
    {
      "key": "/",
      "ifMode": "math",
      "command": null
    },
```

And in `inlineShortcuts` section:
```json
    "//": "\\frac{#?}{#?}",
```

This will disable the default behavior, and when you need fractions, just hit `//` to insert `\frac{}{}`. No implicit arguments.

**Comment by chrostino** [comment]
## System of equation inline shortcut
in `inlineShortuts`:
```
"system": "\\begin{cases} #? \\\\\n #? \\end{cases}",
```
(replace `system` with whichever shortcut suits you)
result:
![Screenshot_06_02_24_16:59](https://github.com/AllanChain/logseq-live-math/assets/100096435/110aeabf-648c-40b8-ae32-d6e937d57aea)
more rows can be added using the right click menu of the extension

**Comment by mufasachan** [comment]
### Toggle text mode (`Ctrl`+`t`)
In `keybinding` list add those two dicts:
```json
    {
      "key": "ctrl+t",
      "ifMode": "math",
      "command": [
        "switchMode",
        "text"
      ]
    },
    {
      "key": "ctrl+t",
      "ifMode": "text",
      "command": [
        "switchMode",
        "math"
      ]
    },
```

Note: The conversion of text mode to math mode might be a little buggy for selection. Usually, redoing the selection or using `ctrl`+`t` again solve the buggy behaviour.

From [CortexJS](https://cortexjs.io/mathfield/guides/shortcuts#keybindings) documentation and some adaptations.
